Packs/Venafi/Integrations/VenafiV2/VenafiV2_test.py

```diff
@@ -0,0 +1,182 @@
+import json
+from freezegun import freeze_time
+
+MOCK_BASEURL = "https://test.com"
+MOCK_CLIENT_ID = "example_client_id"
+MOCK_CLIENT_PASSWORD = "example_password"
+MOCK_USERNAME = "example_username"
+
+
+def util_load_json(path):
+    with open(path, encoding='utf-8') as f:
+        return json.loads(f.read())
+
+
+"""*****TEST LOGIN****"""
+
+
+def test_login_first_time_token_creation(mocker):
+    """
+    Given: An empty integration context
+    When: Login is called for the first time
+    Then: Create a new token and save it to the integration context
+    """
+
+    from VenafiV2 import Client
+
+    mocker.patch("VenafiV2.get_integration_context", return_value={})
+    mocker.patch.object(Client, '_http_request', return_value={
+        "access_token": "access_token",
+        "refresh_token": "refresh_token",
+        "expires_in": 7775999,
+        "expires": 1721806543,
+        "token_type": "Bearer",
+        "scope": "certificate",
+        "identity": "local:{identity}",
+        "refresh_until": 1745566543
+    })
+
+    client = Client(
+        base_url=MOCK_BASEURL,
+        verify=True,
+        username=MOCK_USERNAME,
+        password=MOCK_CLIENT_PASSWORD,
+        client_id=MOCK_CLIENT_ID,
+        proxy=False
+    )
+
+    assert client.token == "access_token"
+
+
+@freeze_time("2024-04-25 00:00:00")
+def test_login_with_valid_token(mocker):
+    """
+    Given: A token in the integration context with a valid expiration time
+    When: Login is called with a valid token
+    Then: Fetch the token from the integration context and log in
+    """
+
+    from VenafiV2 import Client
+
+    mocker.patch("VenafiV2.get_integration_context", return_value={
+        "token": "access_token",
+        "expires": "1715032135"
+    })
+
+    client = Client(
+        base_url=MOCK_BASEURL,
+        verify=True,
+        username=MOCK_USERNAME,
+        password=MOCK_CLIENT_PASSWORD,
+        client_id=MOCK_CLIENT_ID,
+        proxy=False
+    )
+
+    assert client.token == "access_token"
+
+
+@freeze_time("2024-04-25 00:00:00")
+def test_login_with_invalid_token_refresh_required(mocker):
+    """
+    Given: A token in the integration context with an expired expiration time
+    When: Login is called with an invalid token
+    Then: Request a refresh token and save it to the integration context
+    """
+
+    from VenafiV2 import Client
+
+    mocker.patch("VenafiV2.get_integration_context", return_value={
+        "token": "access_token",
+        "expires": "1615032135",
+        "refresh_token": "refresh_token"
+    })
+
+    mocker.patch.object(Client, '_http_request', return_value={
```

Comment:
This is fine, but for your knowledge, you could use also `requests_mock` to mock the exact request made.